azimuth/modules/model_performance/confusion_matrix.py

```diff
@@ -34,26 +41,58 @@ def compute_on_dataset_split(self) -> List[ConfusionMatrixResponse]:  # type: ig
         ds_mng = self.get_dataset_split_manager()
         num_classes = ds_mng.get_num_classes()
         class_ids = list(range(num_classes))
+        class_names = ds_mng.get_class_names()
+        rejection_idx = ds_mng.rejection_class_idx
+
         cf = confusion_matrix(
             y_true=labels,
             y_pred=predictions,
             labels=class_ids,
             normalize="true" if self.mod_options.cf_normalized else None,
         )
 
-        class_names = ds_mng.get_class_names()
+        # Reorder rows and columns so the bandwidth of the matrix is smaller
+        if not self.mod_options.cf_preserved_class_order:
+            # Get a normalized confusion matrix if not already computed
+            if not self.mod_options.cf_normalized:
+                cf_normalized = confusion_matrix(
+                    y_true=labels, y_pred=predictions, labels=class_ids, normalize="true"
+                )
+            else:
+                cf_normalized = cf
 
-        # Put the rejection class last for the confusion matrix
-        rejection_idx = ds_mng.rejection_class_idx
-        if rejection_idx != num_classes - 1:
-            new_order = class_ids[:rejection_idx] + class_ids[rejection_idx + 1 :] + [rejection_idx]
-            cf = cf[np.ix_(new_order, new_order)]
-            class_names = [class_names[i] for i in new_order]
+            # Remove the rejection class so it doesn't influence the algorithm
+            cf_no_rejection = np.delete(
+                np.delete(cf_normalized, rejection_idx, 0), rejection_idx, 1
+            )
+
+            # Get order based on reverse_cuthill_mckee algorithm
+            order_no_rejection = reverse_cuthill_mckee(
+                csr_matrix(cf_no_rejection >= MIN_CONFUSION_CUTHILL_MCKEE)
+            )
+
+            # Get right class indices and add rejection_idx at the end.
```

Comment:
I wasn't sure what "right" meant at first, and thought it would be helpful to indicate that it was the indices for the reduced-bandwidth matrix - but there may be a better way to word it than how I suggested!
```suggestion
            # Get class indices by reduced-bandwidth order and add rejection_idx at the end.
```

Reply:
Could take this approach, too:
```suggestion
            # Get class indices (ordered for plotting) and add rejection_idx at the end.
```